.github/workflows/pull-request-bot.yml

```diff
@@ -139,14 +137,103 @@ jobs:
     - name: Upload doc preview to S3
       run: |
         aws s3 cp target/doc "s3://${S3_BUCKET_NAME}/docs/${{ github.event.pull_request.head.sha }}" --recursive
-    - uses: actions/github-script@v5
-      # NOTE: if comments on each commit become bothersome, add a check that github.event.pull_request.action == "opened"
+
+  generate-server-benchmark:
+    name: Generate server benchmark
+    runs-on: ubuntu-latest
+    outputs:
+      bot-message: ${{ steps.run-benchmark.outputs.bot-message }}
+    steps:
+    - name: Checkout PR
+      uses: actions/checkout@v3
+      with:
+        path: pull-request
+    - name: Checkout origin/main
+      uses: actions/checkout@v3
+      with:
+        repository: awslabs/smithy-rs
+        path: origin-main
+        ref: main
+    - name: Checkout wrk
+      uses: actions/checkout@v3
+      with:
+        repository: wg/wrk
+        path: wrk-build
+        ref: 4.2.0
+    - uses: actions/cache@v2
+      name: Gradle Cache
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+        restore-keys: |
+          ${{ runner.os }}-gradle-
+        # Pinned to the commit hash of v1.3.0
+    - name: Rust Cache
+      uses: Swatinem/rust-cache@842ef286fff290e445b90b4002cc9807c3669641
+      with:
+        sharedKey: ${{ runner.os }}-${{ env.rust_version }}-${{ github.job }}
+        target-dir: ./target
+    - name: Set up JDK
+      uses: actions/setup-java@v1
+      with:
+        java-version: ${{ env.java_version }}
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ env.rust_version }}
+        components: ${{ env.rust_toolchain_components }}
+        default: true
+    - name: Install benchmarks dependencies
+      run: sudo apt-get update && sudo apt-get install -y ${{ env.apt_dependencies }}
+      # Ubuntu 20.04 doesn't have wrk packaged, hence we need to build it 🤦
+      # This will go away as soon as GitHub supports Ubuntu 21.10.
```

Comment:
Would you mind creating an issue for this and linking it here?

Reply:
https://github.com/awslabs/smithy-rs/issues/1310